### PR TITLE
Make k8s keychain vs default keychain configurable

### PR DESCRIPTION
**Pull request by marcofranssen**
https://github.com/philips-labs/slsa-provenance-action/releases/tag/v0.7.0-rc

https://github.com/philips-labs/slsa-provenance-action/actions/runs/1824661573

### CODE REVIEW

File: cmd/slsa-provenance/cli/container.go [hunk 1]
```diff
@@ -53,7 +53,7 @@ func OCI() *cobra.Command {
 				return err
 			}
 
-			opts := oci.WithDefaultClientOptions(cmd.Context(), true)
```
Comment (marcofranssen) [inline cmd/slsa-provenance/cli/container.go:56]: originally we always took the MultKeychain approach. Now that has to be explicitly enabled commandline.

File: cmd/slsa-provenance/cli/options/oci.go [hunk 2]
```diff
@@ -39,4 +46,12 @@ func (o *OCIOptions) AddFlags(cmd *cobra.Command) {
 	cmd.PersistentFlags().StringVar(&o.Repository, "repository", "", "The repository of the oci artifact.")
 	cmd.PersistentFlags().StringVar(&o.Digest, "digest", "", "The digest for the oci artifact.")
 	cmd.PersistentFlags().StringSliceVar(&o.Tags, "tags", []string{"latest"}, "The given tags for this oci release.")
+	cmd.Flags().BoolVar(&o.AllowInsecure, "allow-insecure", false, "whether to allow insecure connections to registries. Don't use this for anything but testing")
+	cmd.Flags().BoolVar(&o.KubernetesKeychain, "k8s-keychain", false, "whether to use the kubernetes keychain instead of the default keychain (supports workload identity).")
```
Comment (marcofranssen) [inline cmd/slsa-provenance/cli/options/oci.go:50]: by default MultiKeychain is disabled now.

**Comment by Brend-Smits** [comment]
I have been testing this in a private registry, but unfortunately, I don't think this is working yet.

**Comment by Brend-Smits** [comment]
Unfortunately, I think this is not yet resolved, internal testing shows that it is still an issue. This can either be in my workflow (am I missing some step), or in slsa action.
Thoughts?
https://github.com/philips-internal/internal-test-slsa-provenance-action

**Comment by marcofranssen** [comment]
The rootcause of our problem is that we are running the `slsa-provenance` binary inside a Docker container in the action. Therefore the binary doesn't know about the docker login that happens outside of the container in a workflow.

We have 2 option here. 

- Docker credentials passed to the action as inputs
- Only install the binary

File: install-slsa-provenance.sh [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+else
+  alias log_info="echo \"INFO:\""
+  alias log_error="echo \"ERROR:\""
+  alias log_warning="echo \"WARN:\""
+fi
+
+set -e
+
+# default to relative path if INSTALL_PATH is not set
+INSTALL_PATH=${INSTALL_PATH:-$(realpath ./.slsa-provenance)}
```
Comment (Brend-Smits) [inline install-slsa-provenance.sh:17]: For some reason my MacOS did not have a ``realpath`` command in ZSH or Bash(3.2). This was no issue on Ubuntu or my Windows machine. Not sure what is going on here.

**Comment by marcofranssen** [comment]
https://github.com/philips-labs/slsa-provenance-action/actions/runs/1840388657

It now also works for draft releases.

**Comment by marcofranssen** [comment]
resolves #94

**Review by Brend-Smits — APPROVED** [review]

File: install-slsa-provenance.sh (hunk 3 above)
Comment (marcofranssen) [reply install-slsa-provenance.sh:20]: MacOS has an old version of Bash by default

```shell
brew update
brew install bash
```

**Pull request merged**